### ISSUE TITLE
ERROR Read timed out

### ISSUE THREAD

**Issue by wojoeraf** [alda-lang/alda#170]
Hey, nice job with alda.

I'm using it on windows with v1.0.0-rc2.
I'm editing an alda file with my editor and listening to it with `alda play --file test.alda`.
But after a bunch of `alda play --file test.alda` I get the Error message: `ERROR Read timed out` and I have to restart alda with `alda restart`.

Do I have a wrong workflow? Is there any other way, to listen to the last edit?

**Comment by daveyarwood** [comment]
Thanks for reporting this. I've actually noticed the same problem intermittently. I'm not sure what the issue is at this point, but I'll keep this issue open until we figure it out. Hopefully we can fix it before the release of Alda 1.0.0.

(Your workflow is good -- that's pretty much exactly what I had in mind for how to use Alda to work on a score. :+1: )

One thing I've noticed is that when my Alda server gets into this weird state, (at least for me) I am able to run some commands, but not others. (`alda restart` works, for example. If I remember correctly, `alda status` and `alda info` work as well.) Is it the same for you?

**Comment by daveyarwood** [comment]
I'm able to reproduce this error by taking the following steps:

- Start an Alda server and leave it running.
- Download an older version of Alda.
- Use the older version to run `alda restart`.

It seems like the problem may be related to connection pooling in the Java HTTP client library we're using. I don't know too much about this... will have to research. In the meantime, this bug is up for grabs if someone has a fix!

**Comment by xyproto** [comment]
I also have this problem.

* Alda client version: 1.0.0-rc14
* Alda server version: 1.0.0-rc14
* 64-bit Arch Linux

```
openjdk version "1.8.0_77"
OpenJDK Runtime Environment (build 1.8.0_77-b03)
OpenJDK 64-Bit Server VM (build 25.77-b03, mixed mode)
```

I have installed the fluidr3 soundfont with the installation script.

The error happens every time I play one of the example files:

```
examples% alda play --file bach_cello_suite_no_1.alda
[27713] Starting Alda server...
[27713] Server up ✓
[27713] ERROR Read timed out
```

The music plays just fine in the background after this error occurs.

**Comment by daveyarwood** [comment]
I've figured out the problem -- in the client code, [we set a 5000 ms socket timeout](https://github.com/alda-lang/alda/blob/master/client/src/alda/AldaServer.java#L66), which means 5000 ms is as long as the command-line client will wait for the server to respond that it's evaluated the score and started playing it. This is OK as long as the process of parsing and evaluating the score takes less than 5 seconds, but for longer scores (the Bach cello suite example score is a good example), it can take 5 seconds or more, which triggers the timeout error.

I don't think we particularly need the client to have a socket timeout, so I have a fix on the way that removes it.
